server/broadcast.go

```diff
@@ -473,34 +479,62 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) ([]string, erro
 		sv = verification.NewSegmentVerifier(Policy)
 	}
 
-	for i := 0; i < MaxAttempts; i++ {
-		// if fails, retry; rudimentary
-		var urls []string
-		if urls, err = transcodeSegment(cxn, seg, name, sv); err == nil {
-			return urls, nil
+	var (
+		startTime = time.Now()
+		attempts  []data.TranscodeAttemptInfo
+		urls      []string
+	)
+	for len(attempts) < MaxAttempts {
+		// if transcodeSegment fails, retry; rudimentary
+		var info data.TranscodeAttemptInfo
+		urls, info, err = transcodeSegment(cxn, seg, name, sv)
+		attempts = append(attempts, info)
+		if err == nil {
+			break
 		}
 
 		if shouldStopStream(err) {
 			glog.Warningf("Stopping current stream due to err=%v", err)
 			rtmpStrm.Close()
-			return nil, err
+			break
 		}
-
 		if isNonRetryableError(err) {
 			glog.Warningf("Not retrying current segment nonce=%d seqNo=%d due to non-retryable error err=%v", nonce, seg.SeqNo, err)
-			return nil, err
+			break
 		}
-
 		// recoverable error, retry
 	}
-	if err != nil {
+	if MetadataQueue != nil {
```

Comment:
Would be great to have a small unit test to verify the `MetadataQueue.Publish` call is being called with the data that we expect.

Reply:
Yeah, will do! Sorry haven't stop to implement the tests yet, but will do before merging (probably todaty)

Reply:
Done!